web/src/utils/validate.ts

```diff
@@ -43,3 +57,139 @@ export const validatePhoneNumber = (phoneNumber: string): boolean => {
   const phoneRegex = /^01(?:0|1|[6-9])(?:\d{3}|\d{4})\d{4}$/;
   return phoneRegex.test(phoneNumber);
 };
+
+/**
+ * Validate start date is same or later than current date by minutes
+ *
+ * @param {Date} startDate Start date
+ * @param {Date} currentDate Current date
+ * @returns {boolean} Result of validation
+ */
+export const validateStartDate = (startDate: Date, currentDate: Date): boolean => {
+  const dateDiff = startDate.getTime() - currentDate.getTime();
+
+  if (Math.abs(dateDiff) < 60000) {
+    if (startDate.getMinutes() >= currentDate.getMinutes()) return true;
+    return false;
+  }
+
+  if (dateDiff < 0) return false;
+
+  return true;
+};
+
+/**
+ * Validate end date is least 1 minute later than start date
+ *
+ * @param {Date} startDate Start date
+ * @param {Date} endDate End date
+ * @param {number} minPeriod Minimum survey period by minutes
+ * @returns {boolean} Result of validation
+ */
+export const validateEndDate = (startDate: Date, endDate: Date, minPeriod: number): boolean => {
+  const dateDiff = endDate.getTime() - startDate.getTime();
+
+  if (dateDiff <= 0) return false;
+
+  if (dateDiff < 60000 * minPeriod) return false;
+
+  return true;
+};
+
+/**
+ * Validate survey data
+ * 1) Check all inputs are filled
+ * 2) Validate start date
+ * 3) Validate end date
+ * 4) Check survey contain at least 1 question
+ * 5) Check selective question contain at least 1 option
+ *
+ * @param {SurveyCreateRequest} surveyData Survey data
+ * @returns {InputCheckResult} Validate result message and error question index(-1 means SurveyDataForm)
+ */
+export const validateSurveyData = (surveyData: SurveyCreateRequest): InputCheckResult => {
```

Comment:
Yeah thank you for separating this but still you also need to separate the logic explicitly. Let me handle this later.